### PR TITLE
[clang][NFC] Refactor `clang/test/SemaCXX/type-traits.cpp` to use modern `static_assert`

### PR DESCRIPTION
**Pull request by AMP999**
static_assert is supported in all modes, so the T/F macros aren't needed anymore.

### CODE REVIEW

File: clang/test/SemaCXX/type-traits.cpp [hunk 1]
```diff
@@ -2671,212 +2660,212 @@ struct has_unique_object_representations {
   static const bool value = __has_unique_object_representations(T);
 };
```
Comment (cor3ntin) [inline clang/test/SemaCXX/type-traits.cpp:2661]: This is no longer needed at all

File: clang/test/SemaCXX/type-traits.cpp [hunk 2]
```diff
@@ -214,56 +212,56 @@ struct HasVirtBase : virtual ACompleteType {};
...
-  { int arr[F(__is_pod(NonPODArNB))]; }
-  { int arr[F(__is_pod(void))]; }
-  { int arr[F(__is_pod(cvoid))]; }
-// { int arr[F(__is_pod(NonPODUnion))]; }
-
-  { int arr[T(__is_pod(ACompleteType))]; }
-  { int arr[F(__is_pod(AnIncompleteType))]; } // expected-error {{incomplete type}}
-  { int arr[F(__is_pod(AnIncompleteType[]))]; } // expected-error {{incomplete type}}
-  { int arr[F(__is_pod(AnIncompleteType[1]))]; } // expected-error {{incomplete type}}
+  static_assert(__is_pod(int), "");
```
Comment (cor3ntin) [inline clang/test/SemaCXX/type-traits.cpp:215]: Maybe we should compile with `-Wno-c++17-extensions` and get rid of the empty message

**Review by cor3ntin — COMMENTED** [review]
Looks like a nice improvement.
I have a couple of suggestions

File: clang/test/SemaCXX/type-traits.cpp (hunk 1 above)
Comment (erichkeane) [reply clang/test/SemaCXX/type-traits.cpp:2661]: The intent here was to emulate how the library would use the builtin as a type-trait.  

Also, see this bug here: https://github.com/llvm/llvm-project/issues/77585 for additional comments.

Comment (cor3ntin) [reply clang/test/SemaCXX/type-traits.cpp:2661]: Except we don't do that for any other trait in that file.

Comment (erichkeane) [reply clang/test/SemaCXX/type-traits.cpp:2661]: Yep, thats true too, it was the intent at the time though (and to better emulate hte test cases we were given, note that the whole section is basically a copy/paste + de-c++17'izing of what STL gave me).

My comment on the bug has more context.

File: clang/test/SemaCXX/type-traits.cpp (hunk 2 above)
Comment (AMP999) [reply clang/test/SemaCXX/type-traits.cpp]: @cor3ntin Empty messages have been eliminated by commit `5e043de`. Should we also remove the messages in the `__has_unique_object_representations` section?

**Review by cor3ntin — APPROVED** [review]
LGTM

**Comment by AMP999** [comment]
@cor3ntin All checks are green now. Could you help me land this patch for me?

**Pull request merged**